test/libsolidity/semanticTests/viaYul/array_storage_index_access.sol

```diff
@@ -0,0 +1,54 @@
+contract C {
+	uint[] storageArray;
+	function test_indicies(uint256 len) public
+	{
+		storageArray.length = len;
+
+		for (uint i = 0; i < len; i++)
+			storageArray[i] = i + 1;
+
+		for (uint i = 0; i < len; i++)
+			require(storageArray[i] == i + 1);
+	}
+
+	function test_zeroed_indicies(uint256 len) public
+	{
+		storageArray.length = len;
+
+		for (uint i = 0; i < len; i++)
+			storageArray[i] = i + 1;
+
+		storageArray.length = 0;
```

Comment:
Maybe better to let the length to 3 or something and then check using inline assembly and `sload`, that the slots are zeroed-out. In the loop below, inline assembly is redundant.